### PR TITLE
root option is working (fixed #986)

### LINKED ISSUE #986: root attribute ignored
```ruby
render json: @current_user, root: 'resource', adapter: :json
```
gives output 
```json
{
  "user": {
    "id": 1,
    "name": "Rambo"
  }
}
```
when expected 
```json
{
  "resource": {
    "id": 1,
    "name": "Rambo"
  }
}
```

### PR DESCRIPTION
**Pull request by mareczek**
Will be very grateful for a quick resolution.

### CODE REVIEW

File: test/serializers/root_test.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+require 'test_helper'
+
+module ActiveModel
+  class Serializer
+    class RootTest < Minitest::Test
+
+      def setup
+        @profile = Profile.new({ name: 'Name 1', description: 'Description 1', comments: 'Comments 1' })
+        @profile_serializer = ProfileSerializer.new(@post, {root: 'smth'})
```
Comment (bf4) [inline test/serializers/root_test.rb]: These tests are testing the value of Serializer#json_key, but the issue in #986 was

```ruby
render json: @current_user, root: 'resource', adapter: :json
```

Which uses the full stack of something like `JsonAdapter.create(ProfileSerializer.new(@post, root: 'smth')).serializable_hash`.  I'm not sure you just want to test `ProfileSerializer.new(@post, root: 'smth').json_key`.

That said, the behavior you desire was specifically removed from AMS in #958 

```diff
# controller
-    ADAPTER_OPTION_KEYS = [:include, :fields, :root, :adapter]
+    ADAPTER_OPTION_KEYS = [:include, :fields, :adapter]
# serializer
     def initialize(object, options = {})

-      @root       = options[:root] || (self.class._root ? self.class.root_name : false)
+      @root       = options[:root]

    def json_key
-      if root == true || root.nil?
-        self.class.root_name
-      else
-        root
-      end
+      self.class.root_name
     end
 # adapter

       def root
-        @options.fetch(:root) { serializer.json_key }
+        serializer.json_key.to_sym if serializer.json_key
       end
 
       def include_meta(json)
-        json[meta_key] = meta if meta && root
+        json[meta_key] = meta if meta
         json
       end
```

I brought up documenting these changes in #966 but it hasn't been merged/finalized yet

Comment (joaomdmoura) [inline test/serializers/root_test.rb]: :+1: Indeed. We could bring some of the older tests to validate it.

**Comment by joaomdmoura** [comment]
@mareczek could you check @bf4 comment? Maybe check #958, bring some tests back to this PR.

**Comment by vyrak** [comment]
One thing to note is that the root option is ignored when serializing an empty array.

**Comment by vyrak** [comment]
@joaomdmoura 

The PR ported some tests from https://github.com/joaomdmoura/active_model_serializers/commit/329691276a1bb2827f20bceda5d15fd79a08d3dc. Let me know if there are other tests that need to be ported back.

**Comment by joaomdmoura** [comment]
Tks @vyrak, awesome, I'm closing this one in favor of #1013

**Pull request closed without merging**